src/libraries/System.Web.HttpUtility/src/System/Web/Util/HttpEncoder.cs

```diff
@@ -233,21 +241,14 @@ internal static string JavaScriptStringEncode(string? value)
                     {
                         // valid 2 hex chars
                         b = (byte)((h1 << 4) | h2);
-                        i += 2;
+                        pos += 2;
                     }
                 }
 
                 decodedBytes[decodedBytesCount++] = b;
             }
 
-            if (decodedBytesCount < decodedBytes.Length)
-            {
-                byte[] newDecodedBytes = new byte[decodedBytesCount];
-                Array.Copy(decodedBytes, newDecodedBytes, decodedBytesCount);
-                decodedBytes = newDecodedBytes;
-            }
-
-            return decodedBytes;
+            return decodedBytes.Slice(0, decodedBytesCount).ToArray();
```

Comment:
@MihaZupan In the original code we do not allocate twice if there are no '%' character.
Should I set a flag if count > 512 and we didn't found any special char - to return already allocated new byte[count]?